common/stoplight.hpp

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <mutex>
+#include <chrono>
+#include <condition_variable>
+#include "phonebook.hpp"
+
+namespace ILLIXR {
+
+/**
+ * @brief A boolean condition-variable.
+ *
+ * Inspired by https://docs.python.org/3/library/threading.html#event-objects
+ */
+class Event {
+private:
+	mutable std::mutex _m_mutex;
+	mutable std::condition_variable _m_cv;
+	bool _m_value = false;
+
+public:
+
+	/**
```

Comment:
Thoughts on using [noexcept](https://stackoverflow.com/questions/10787766/when-should-i-really-use-noexcept) for our classes?

Reply:
If we later add an exception, then we would have to change the interface. I think its best to leave the option open unless there is a performance or semantic reason to do so.

Reply:
Agreed about changing the interface. Thinking about this more, any implementation change which adds the possibility of an exception being thrown is a change to the interface, so we could make an argument that forcing these functions to `std::terminate` is more stable (i.e. introduced exceptions must be handled).